### PR TITLE
Add requests app ECS & LB

### PR DESCRIPTION
**Pull request by jamieparkinson**
This is the first part of the requests app infra, most of the diff is [this module](https://github.com/wellcomecollection/catalogue-api/tree/main/terraform/modules/service) copied and pasted from the catalogue-api repo.

### CODE REVIEW

**Comment by kenoir** [comment]
Don't forget to ignore the deployment label tag as in: https://github.com/wellcomecollection/catalogue-api/blob/main/terraform/catalogue_api/provider.tf#L8

File: infra/modules/ecs_service/variables.tf [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+variable "secrets" {
+  type    = map(string)
+  default = {}
+}
+
+variable "container_port" {
+  type    = number
+  default = 9001
+}
```
Comment (alexwlchan) [inline infra/modules/ecs_service/variables.tf:22]: Nit: do we want to set a default here? Feels like there's an easy mistake where you forget to set it and wonder why you don't see anything.

File: infra/scoped/ecs_services.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+module "requests" {
+  source = "../modules/ecs_service"
+
+  service_name            = "requests-${terraform.workspace}"
+  deployment_service_name = "requests"
+
+  container_image        = "${local.requests_repository}:env.${terraform.workspace}"
+  container_port         = local.requests_container_port
+  deployment_service_env = terraform.workspace
+  desired_task_count     = 1
```
Comment (alexwlchan) [inline infra/scoped/ecs_services.tf:14]: Why only 1?

File: infra/scoped/ecs_services.tf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  load_balancer_arn           = aws_lb.identity_api.arn
+  load_balancer_listener_port = local.requests_lb_port
+
+  cluster_arn = aws_ecs_cluster.identity.arn
+  vpc_id      = aws_vpc.main.id
+  subnets     = local.private_subnets
+
+  security_group_ids = [
+    aws_security_group.local.id,
```
Comment (alexwlchan) [inline infra/scoped/ecs_services.tf:40]: What's the local security group? Do we need it?

**Review by alexwlchan — DISMISSED** [review]

File: infra/scoped/ecs_services.tf (hunk 3 above)
Comment (jamieparkinson) [reply infra/scoped/ecs_services.tf:40]: It's what we have called `ingress` elsewhere - local in the sense that it allows ingress from within the VPC

File: infra/scoped/ecs_services.tf (hunk 2 above)
Comment (jamieparkinson) [reply infra/scoped/ecs_services.tf:14]: true, I'll make this the same as how we do tasks and AZs in the catalogue API

**Comment by jamieparkinson** [comment]
(these tests are failing at the moment because I've bumped a module version pre-emptively, cf https://github.com/wellcomecollection/terraform-aws-ecs-service/pull/34)

File: infra/modules/ecs_service/main.tf [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  source    = "git::github.com/wellcomecollection/terraform-aws-ecs-service.git//modules/secrets?ref=v3.6.2"
+  secrets   = var.secrets
+  role_name = module.task_definition.task_execution_role_name
+}
+
+module "task_definition" {
+  source = "git::github.com/wellcomecollection/terraform-aws-ecs-service.git//modules/task_definition?ref=v3.6.2"
+
+  cpu    = 1024
+  memory = 2048
```
Comment (alexwlchan) [inline infra/modules/ecs_service/main.tf:43]: Minor: worth making these variables?

File: infra/modules/ecs_service/target_group.tf [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  target_type = "ip"
+
+  protocol = "TCP"
+  port     = module.nginx_container.container_port
+  vpc_id   = var.vpc_id
+
+  # The default deregistration delay is 5 minutes, which means that ECS
+  # takes around 5–7 mins to fully drain connections to and deregister
+  # the old task in the course of its blue/green. deployment of an
```
Comment (alexwlchan) [inline infra/modules/ecs_service/target_group.tf:14]: nit:

```suggestion
  # the old task in the course of its blue/green deployment of an
```

**Review by alexwlchan — APPROVED** [review]

**Pull request merged**